File: SEP3_Database/src/main/java/com/example/SEP3_Database/model/domain/list/userList/FriendList.java
package com.example.SEP3_Database.model.domain.list.userList;

import com.example.SEP3_Database.model.domain.unit.user.Friend;

import java.util.ArrayList;

public class FriendList {
    private ArrayList<Friend> Friends;

    public FriendList()
    {
        Friends = new ArrayList<>();
    }

    public String addNewFriend(Friend newFriend)
    {
        if (getFriendById(newFriend.getId())!=null)
        {
            return "Friend [" + newFriend.getId() + "] is not a new friend.";
        }
        Friends.add(newFriend);
        return null;
    }

    public int getSize()
    {
        return Friends.size();
    }

    public Friend getFriendByIndex(int index)
    {
        if (index>=0&&index<Friends.size())
        {
            return Friends.get(index);
        }
        else
        {
            return null;
        }
    }

    public Friend getFriendById(String id)
    {
        for (int x = 0; x< Friends.size(); x++)
        {
            if (Friends.get(x).getId().equals(id))
            {
                return Friends.get(x);
            }
        }
        return null;
    }

    public String updateFriend(Friend account)
    {
        for (int x = 0; x< Friends.size(); x++)
        {
            if (Friends.get(x).getId().equals(account.getId()))
            {
                Friends.get(x).update(account);
                return null;
            }
        }
        return "Can't find friend [" + account.getId() + "].";
    }

    public void removeFriendById(String id)
    {
        for (int x = 0; x< Friends.size(); x++)
        {
            if (Friends.get(x).getId().equals(id))
            {
                Friends.remove(x);
                break;
            }
        }
    }

    public FriendList copy()
    {
        FriendList copy = new FriendList();
        for (int x = 0; x< Friends.size(); x++)
        {
            copy.addNewFriend(Friends.get(x).copy());
        }
        return copy;
    }
}

File: SEP3_PostSystem/Model/List/UserList/FriendList.cs
using System.Collections.Generic;
using SEP3_PostSystem.Model.Unit.User;

namespace SEP3_PostSystem.Model.List.UserList
{
    public class FriendList
    {
        public List<Friend> Friends { get; set; }

        public FriendList()
        {
            Friends = new List<Friend>();
        }

        public string AddNewFriend(Friend newFriend)
        {
            if (newFriend!=null)
            {
                if (GetFriendById(newFriend.GetId())!=null)
                {
                    return "Friend [" + newFriend.GetId() + "] is not a new friend.";
                }
                Friends.Add(newFriend);
                return null;
            }
            return "Input null.";
        }

        public Friend GetFriendById(string id)
        {
            foreach (var friend in Friends)
            {
                if (friend.GetId().Equals(id))
                {
                    return friend;
                }
            }

            return null;
        }

        public string UpdateFriend(Friend newFriend)
        {
            foreach (var friend in Friends)
            {
                if (friend.GetId().Equals(newFriend.GetId()))
                {
                    friend.Update(newFriend);
                    return null;
                }
            }

            return "Can't find friend [" + newFriend.GetId() + "].";
        }

        public void RemoveFriendById(string id)
        {
            foreach (var friend in Friends)
            {
                if (friend.GetId().Equals(id))
                {
                    Friends.Remove(friend);
                }
            }
        }
        
        public FriendList Copy()
        {
            FriendList copy = new FriendList();
            foreach (var friend in Friends)
            {
                copy.Friends.Add(friend.Copy());
            }
            return copy;
        }
    }
}